src/extension.ts

```diff
@@ -847,6 +849,28 @@ function createMonitor(): any {
     });
 }
 
+function createIdfTerminal() {
+    PreCheck.perform([webIdeCheck, openFolderCheck], () => {
+        const envSetCmd = process.platform === "win32" ? "set" : "export";
+        const modifiedEnv = utils.appendIdfAndToolsToPath();
+        const espIdfTerminal = vscode.window.createTerminal({ name: "ESP-IDF Terminal", env: modifiedEnv,
+                                                              cwd: workspaceRoot.fsPath });
+        espIdfTerminal.show();
+        espIdfTerminal.sendText(`${envSetCmd} IDF_PATH=${modifiedEnv.IDF_PATH}`);
+        let IDF_ADD_PATHS_EXTRAS =
+            `${path.join(modifiedEnv.IDF_PATH, "components", "esptool_py", "esptool")}`;
+        IDF_ADD_PATHS_EXTRAS =
+            `${IDF_ADD_PATHS_EXTRAS}${path.delimiter}${path.join(modifiedEnv.IDF_PATH,
+                                                                 "components", "espcoredump")}`;
+        IDF_ADD_PATHS_EXTRAS =
+            `${IDF_ADD_PATHS_EXTRAS}${path.delimiter}${path.join(modifiedEnv.IDF_PATH,
+                                                                 "components", "partition_table")}`;
```

Comment:
Note that these paths may change between IDF versions. This is probably okay for now, but we need to find some way to communicate such path to the IDEs. Have logged this as IDF-1501.